src/libraries/System.Private.CoreLib/src/System/UInt128.cs

```diff
@@ -0,0 +1,1654 @@
+
+        /// <inheritdoc cref="IEqualityOperators{TSelf, TOther}.op_Inequality(TSelf, TOther)" />
+        public static bool operator !=(UInt128 left, UInt128 right) => (left._lower != right._lower) || (left._upper != right._upper);
+
+        //
+        // IIncrementOperators
+        //
+
+        /// <inheritdoc cref="IIncrementOperators{TSelf}.op_Increment(TSelf)" />
+        public static UInt128 operator ++(UInt128 value) => value + One;
+
+        /// <inheritdoc cref="IIncrementOperators{TSelf}.op_CheckedIncrement(TSelf)" />
+        public static UInt128 operator checked ++(UInt128 value) => checked(value + One);
+
+        //
+        // IMinMaxValue
+        //
+
+        /// <inheritdoc cref="IMinMaxValue{TSelf}.MinValue" />
+        public static UInt128 MinValue => new UInt128(0, 0);
+
+        /// <inheritdoc cref="IMinMaxValue{TSelf}.MaxValue" />
+        public static UInt128 MaxValue => new UInt128(0xFFFF_FFFF_FFFF_FFFF, 0xFFFF_FFFF_FFFF_FFFF);
+
+        //
+        // IModulusOperators
+        //
+
+        /// <inheritdoc cref="IModulusOperators{TSelf, TOther, TResult}.op_Modulus(TSelf, TOther)" />
+        public static UInt128 operator %(UInt128 left, UInt128 right)
+        {
+            UInt128 quotient = left / right;
+            return left - (quotient * right);
+        }
+
+        //
+        // IMultiplicativeIdentity
+        //
+
+        /// <inheritdoc cref="IMultiplicativeIdentity{TSelf, TResult}.MultiplicativeIdentity" />
+        static UInt128 IMultiplicativeIdentity<UInt128, UInt128>.MultiplicativeIdentity => One;
+
+        //
+        // IMultiplyOperators
+        //
+
+        /// <inheritdoc cref="IMultiplyOperators{TSelf, TOther, TResult}.op_Multiply(TSelf, TOther)" />
+        public static UInt128 operator *(UInt128 left, UInt128 right)
+        {
+            ulong upper = Math.BigMul(left._lower, right._lower, out ulong lower);
+            upper += (left._upper * right._lower) + (left._lower * right._upper);
+            return new UInt128(upper, lower);
+        }
+
+        /// <inheritdoc cref="IMultiplyOperators{TSelf, TOther, TResult}.op_CheckedMultiply(TSelf, TOther)" />
+        public static UInt128 operator checked *(UInt128 left, UInt128 right)
+        {
+            UInt128 upper = BigMul(left, right, out UInt128 lower);
+
+            if (upper != 0U)
+            {
+                ThrowHelper.ThrowOverflowException();
+            }
+
+            return lower;
+        }
+
+        internal static UInt128 BigMul(UInt128 left, UInt128 right, out UInt128 lower)
+        {
+            // Adaptation of algorithm for multiplication
+            // of 32-bit unsigned integers described
+            // in Hacker's Delight by Henry S. Warren, Jr. (ISBN 0-201-91465-4), Chapter 8
+            // Basically, it's an optimized version of FOIL method applied to
+            // low and high dwords of each operand
```

Comment:
Copy + pasted comment, it’s not dwords anymore ;)

Reply:
Fixed.